libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt

```diff
@@ -1,165 +1,95 @@
-        mainCarContext.mapboxCarMap.carMapSurface?.getStyle()?.let { style ->
-            routeLineApi.updateWithRouteProgress(routeProgress) { result ->
-                routeLineView.renderRouteLineUpdate(style, result)
-            }
-            routeArrowApi.addUpcomingManeuverArrow(routeProgress).also { arrowUpdate ->
-                routeArrowView.renderManeuverUpdate(style, arrowUpdate)
+    private fun attachRouteLine(mapboxCarMapSurface: MapboxCarMapSurface) {
+        routeLineComponent = RouteLineComponent(
+            mapboxMap = mapboxCarMapSurface.mapSurface.getMapboxMap(),
+            mapPlugins = mapboxCarMapSurface.mapSurface,
+            options = routeLineOptions
+                ?: MapboxRouteLineOptions.Builder(mapboxCarMapSurface.carContext)
+                    .withRouteLineBelowLayerId("road-label-navigation")
+                    .build(),
+            contractProvider = { contract }
+        ).also { routeLineComponent ->
+            mapboxCarMapSurface.getStyleAsync {
+                MapboxNavigationApp.registerObserver(routeLineComponent)
             }
         }
     }
 
-    private var styleLoadedListener: OnStyleLoadedListener? = null
-
-    constructor(
-        mainCarContext: MainCarContext,
-    ) : this(mainCarContext, NavigationRoutesProvider(mainCarContext.mapboxNavigation))
-
-    override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
-        logAndroidAuto("CarRouteLine carMapSurface loaded $mapboxCarMapSurface")
-        val locationPlugin = mapboxCarMapSurface.mapSurface.location
-        val scope = MainScope().also { styleScope = it }
-        styleLoadedListener = mapboxCarMapSurface.handleStyleOnAttached { style ->
-            val routeLineOptions = getMapboxRouteLineOptions(style)
-            routeLineView = MapboxRouteLineView(routeLineOptions)
-            routeLineApi = MapboxRouteLineApi(routeLineOptions).also {
-                routeLineView.initializeLayers(style)
-            }
-            routeArrowApi = MapboxRouteArrowApi()
-            routeArrowView = MapboxRouteArrowView(
-                RouteArrowOptions.Builder(mainCarContext.carContext)
-                    .withAboveLayerId(TOP_LEVEL_ROUTE_LINE_LAYER_ID)
-                    .build()
-            )
-
-            locationPlugin.addOnIndicatorPositionChangedListener(onPositionChangedListener)
-            mainCarContext.mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
-            routesProvider.registerRoutesListener(routesListener)
-
-            scope.coroutineContext.cancelChildren()
-            scope.launch {
-                mainCarContext.routeAlternativesEnabled.collect { enabled ->
-                    if (enabled) {
-                        routeLineView.showAlternativeRoutes(style)
-                    } else {
-                        routeLineView.hideAlternativeRoutes(style)
-                    }
-                }
```

Comment:
Where is this handled now? Looks like we don't have that logic anymore.

Reply:
Good catch!

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/5912/files#diff-6198cd76f0edf02371f3b1f7d936ab739daddc8f9063c26b62459bd42f8452a2R81

Ok added it back in